libmamba/include/mamba/core/transaction.hpp

```diff
@@ -126,7 +122,7 @@ namespace mamba
             const std::vector<MatchSpec>& specs_to_install,
             MultiPackageCache& caches
         );
-        MTransaction(MSolver& solver, MultiPackageCache& caches);
+        MTransaction(MPool& pool, MSolver& solver, MultiPackageCache& caches);
```

Comment:
Because the member `m_pool` is a value and not a reference, it is not clear what is going on here. In the constructor implementation we pass `m_pool(pool)`,  what does this mean? Is it a copy?

Reply:
It's a shared reference. We had the discussion last autumn to make `MPool` interally based on an `std::shared_ptr`.
This will also change as we refactor the pool.

Reply:
Ah ok, then why not pass it by value? that would make it clear that whatever the model, there is ownership being acquired. There is no condition to that so no reason to pass a reference.

Reply:
I missed your answer before I merged, I can make the change in a mechanical PR.